### ISSUE TITLE
[SLPVectorizerPass] Assertion `isValidOperands(V1, V2, Mask)` after D144958

### ISSUE THREAD

**Issue by vzakhari** [llvm/llvm-project#61395]
93a9be0cea0a4c8da820b455fb279ece09a5ca70 affects CPU2017/521.wrf and CPU2006/481.wrf compiled with Flang.

Small reproducer attached: [reduced.bc.gz](https://github.com/llvm/llvm-project/files/10964027/reduced.bc.gz)

Run command: `opt -passes="function(slp-vectorizer)" reduced.bc`
```console
opt: .../llvm/lib/IR/Instructions.cpp:2137: llvm::ShuffleVectorInst::ShuffleVectorInst(llvm::Value*, llvm::Value*, llvm::ArrayRef<int>, const llvm::Twine&, llvm::Instruction*): Assertion `isValidOperands(V1, V2, Mask) && "Invalid shuffle vector instruction operands!"' failed.
```

@alexey-bataev could you please take a look?

**Comment by alexey-bataev** [comment]
Fixed in 874c49f55454cb285282e6d184f809945c0beca1

**Comment by vzakhari** [comment]
All fixed. Thank you!
